x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -42,45 +43,79 @@ export const useUnifiedSearch = () => {
     to: toTS,
   };
 
+  const queryToString = (query?: Query) =>
+    typeof query?.query !== 'string' ? JSON.stringify(query?.query) : query?.query;
+
+  const handleSelectedTimeRangeChange = useCallback(
+    (selectedTime: { startTime: string; endTime: string; isInvalid: boolean }) => {
+      if (selectedTime.isInvalid) {
+        return;
+      }
+      setSelectedTimeRange(selectedTime);
+    },
+    [setSelectedTimeRange]
+  );
+
   const submitFilterChange = useCallback(
     (query?: Query, dateRange?: TimeRange, filters?: Filter[]) => {
       if (filters) {
-        filterManager.setFilters(filters);
+        applyFilters(filters);
+        filterManager.setFilters([...filterManager.getFilters(), ...filters]);
+      }
+
+      if (dateRange) {
+        handleSelectedTimeRangeChange({
+          startTime: dateRange.from,
+          endTime: dateRange.to,
+          isInvalid: false,
+        });
+        setTime({
+          ...getTime(),
+          ...dateRange,
+        });
       }
 
-      setTime({
-        ...getTime(),
-        ...dateRange,
-      });
+      if (query) {
+        applyFilterQuery({
```

Comment:
should we use `setFilterQuery` here and in other places where we use "apply".  Seems like that's been the pattern when updating state.

Reply:
Yes, I agree, thanks! Renamed ✅